packages/stark-ui/src/modules/pretty-print/services/pretty-print.service.ts

```diff
@@ -0,0 +1,149 @@
+import { STARK_LOGGING_SERVICE, StarkLoggingService } from "@nationalbankbelgium/stark-core";
+import { StarkPrettyPrintFormat } from "../types";
+
+
+/**
+ * The prefix used in the PrismJs CSS classes
+ */
+const prismClassPrefix = "language-";
+
+@Injectable()
+export class StarkPrettyPrintServiceImpl implements StarkPrettyPrintService {
+	public constructor(@Inject(STARK_LOGGING_SERVICE) public logger: StarkLoggingService) {}
+
+	public format(data: string, format: StarkPrettyPrintFormat, highlightingEnabled: boolean): Observable<string> {
+		let prismGrammar: Grammar = <any>"";
+		let prismClass = "";
+		const options: Options = {};
+		
+		switch (format) {
+			case "xml":
+				prismGrammar = Prism.languages["markup"];
+				prismClass = prismClassPrefix + "markup";
+				options.parser = "html";
+				options.htmlWhitespaceSensitivity = "ignore";
+				options.plugins = [ prettierParserHtml ];
+				break;
+			case "html":
+				prismGrammar = Prism.languages["markup"];
+				prismClass = prismClassPrefix + "markup";
+				options.parser = "angular";
+				options.htmlWhitespaceSensitivity = "ignore";
+				options.plugins = [
+					prettierParserAngular,
+					prettierParserEstree,
+					prettierParserBabel,
+					prettierParserHtml,
+					prettierParserPostcss
+				];
+				break;
+			case "json":
+				prismGrammar = Prism.languages["json"];
+				prismClass = prismClassPrefix + format;
+				try {
+					JSON.parse(data);
+				} catch (error: any) {
+					this.logger.warn("Pretty print service ", error);
+					return throwError(data);
+				}
+				options.parser = "json";
+				options.plugins = [
+					prettierParserEstree,
+					prettierParserBabel
+				]
+				break;
+			case "css":
+				prismGrammar = Prism.languages["css"];
+				prismClass = prismClassPrefix + format;
+				options.parser = "css";
+				options.plugins = [
+					prettierParserPostcss
+				];
+				break;
+			case "scss":
+				prismGrammar = Prism.languages["scss"];
+				prismClass = prismClassPrefix + format;
+				options.parser = "scss";
+				options.plugins = [
+					prettierParserPostcss
+				];
+				break;
+			case "sql":
+				prismGrammar = Prism.languages["sql"];
+				prismClass = prismClassPrefix + format;
+				data = sqlFormatter.format(data, { language: "sql" });
+				break;
+			case "javascript":
+				prismGrammar = Prism.languages["javascript"];
+				prismClass = prismClassPrefix + format;
+				options.parser = "babel";
+				options.plugins = [
+					prettierParserEstree,
+					prettierParserBabel
+				];
+				break;
+			case "typescript":
+				prismGrammar = Prism.languages["typescript"];
+				prismClass = prismClassPrefix + format;
+				options.parser = "typescript";
+				options.plugins = [
+					prettierParserEstree,
+					prettierParserTypescript ];
+				break;
+			default:
+				this.logger.warn("PrettyPrintService : Unknown format -> " , format);
+				return throwError(data);
+		}
+
+		// if no parser this means sql or default.
+		if (options.parser) {
```

Comment:
Since `default` returns `throwError(data);`, the only case where Prettier won't be used is when the format is `sql`.

Could you adapt the logic to something similar to:

```typescript
if (this.format === "sql") {
    // sql code currently in the else condition
}

// the else condition is not necessary since the `sql` condition returns something
return from(prettierFormat(data, options)).pipe(
	toArray(),
	map((prettyStringArray: string[]): string => {
		const prettyString = prettyStringArray.join("");
		if (highlightingEnabled) {
			const prismString = Prism.highlight(prettyString, prismGrammar, format);
			return "<pre class='" + prismClass + "'><code class='" + prismClass + "'>" + prismString + "</code></pre>";
		} else {
			return prettyString;
		}
	}),
	catchError((error: any) => {
		this.logger.warn(error);
		return throwError(data);
	})
);
```